### PR TITLE
Fix - Event review

### PR DESCRIPTION
**Pull request by puranban**
Depends on https://github.com/idmc-labs/helix-server/pull/416

Related to
- https://github.com/idmc-labs/helix2.0-meta/issues/172
- https://github.com/idmc-labs/helix2.0-meta/issues/247
- https://github.com/idmc-labs/helix2.0-meta/issues/232

## Changes

* Fix comment type colors 
* Add triangulation Modal
* Add comment in the figure unit field.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments

### CODE REVIEW

**Review by priyesh777 — APPROVED** [review]

**Review by priyesh777 — APPROVED** [review]

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/index.tsx [hunk 1]
```diff
@@ -799,6 +816,12 @@ function NudeEventTable(props: EventsProps) {
                     />
                 </Modal>
             )}
+            {shouldShowEventTriangulationChangeModal && (
+                <TriangulationModal
+                    eventId={editTriangulationEventId?.id ?? ''}
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/index.tsx:821]: Fixed

File: src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/index.tsx [hunk 2]
```diff
@@ -581,11 +596,12 @@ function NudeEventTable(props: EventsProps) {
                     )
                         ? handleAssignYourself
                         : undefined,
+                    onChangeTriangulation: handleTriangulationChange,
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/index.tsx:599]: Fixed!

File: src/components/TrafficLightInput/ReviewComments/CommentForm/index.tsx [hunk 3]
```diff
@@ -134,21 +130,19 @@ const schema: FormSchema = {
 };
 
 const defaultFormValues: PartialForm<FormType> = {
-    commentType: 'GREY',
+    commentType: 'GREEN',
```
Comment (tnagorra) [inline src/components/TrafficLightInput/ReviewComments/CommentForm/index.tsx:133]: This should be conditional.

Comment (tnagorra) [reply src/components/TrafficLightInput/ReviewComments/CommentForm/index.tsx:133]: Fixed!

File: src/components/forms/EventForm/index.tsx [hunk 4]
```diff
@@ -974,15 +973,6 @@ function EventForm(props: EventFormProps) {
                 error={error?.fields?.eventNarrative}
                 readOnly={readOnly}
             />
-            <Switch
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/forms/EventForm/index.tsx:977]: Fixed. Need to check if the event can be created.

File: src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/EventsAction/index.tsx [hunk 5]
```diff
@@ -36,6 +37,7 @@ export interface ActionProps {
     children?: React.ReactNode;
     editLinkRoute?: RouteData;
     editLinkAttrs?: Attrs;
+    onChangeTriangulation?: (id: string) => void;
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/EventsAction/index.tsx:40]: Fixed.

File: src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/TriangulationModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
+import React, { useCallback, useContext, useMemo } from 'react';
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/TriangulationModal/index.tsx:1]: Fixed!

File: src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/EventsAction/index.tsx [hunk 7]
```diff
@@ -185,6 +188,17 @@ function EventActionCell(props: ActionProps) {
...
             )}
+            {onChangeTriangulation && (
+                <QuickActionButton
+                    name={id}
+                    onClick={onChangeTriangulation}
+                    title="triangulation"
+                    disabled={disabled || !onChangeTriangulation}
+                    transparent
+                >
+                    <IoShapesOutline />
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/EventsAction/index.tsx:199]: Fixed

File: src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/index.tsx [hunk 8]
```diff
@@ -552,8 +567,8 @@ function NudeEventTable(props: EventsProps) {
                         && !!eventPermissions?.sign_off
                     ) || (
                         !!figurePermissions?.approve
-                        && !!datum?.assignee
-                        && datum.assignee.id === user?.id
+                            && !!datum?.assignee
+                            && datum.assignee.id === user?.id
```
[the remark this answers is not in the document]
Comment (tnagorra) [reply src/components/tables/EventsEntriesFiguresTable/NudeEventsTable/index.tsx:571]: Fixed

**Pull request merged**